### PR TITLE
Fix clang 9 -Wdefaulted-function-deleted

### PR DESCRIPTION
**Pull request by RAOF**

### CODE REVIEW

**Comment by RAOF** [comment]
clang-9 has started warning about a couple of pieces of Mir code, for example:
```
../include/miral/miral/window_management_options.h:56:5: error: explicitly defaulted default constructor is implicitly deleted [-Werror,-Wdefaulted-function-deleted]
    WindowManagerOptions() = default;
    ^
../include/miral/miral/window_management_options.h:61:44: note: default constructor of 'WindowManagerOptions' is implicitly deleted because field 'policies' of const-qualified type 'const std::vector<WindowManagerOption>' would not be initialized
    std::vector<WindowManagerOption> const policies;
```

It's simple to fix them, so do.

File: include/miral/miral/window_management_options.h [hunk 1]
```diff
@@ -53,7 +53,7 @@ inline auto add_window_manager_policy(std::string const& name, Args&... args) ->
 class WindowManagerOptions
 {
 public:
-    WindowManagerOptions() = default;
+    WindowManagerOptions();
```
Comment (AlanGriffiths) [inline include/miral/miral/window_management_options.h]: This looks like the wrong fix. I suggest:
```c++
    WindowManagerOptions() = delete;
```
as that does not change the meaning of the code.

Comment (RAOF) [reply include/miral/miral/window_management_options.h]: Hah! The warning proves valuable already. I thought the intent of `WindowManagerOptions() = default` was to make a no-parameter constructor available :grin:

Comment (AlanGriffiths) [reply include/miral/miral/window_management_options.h]: That is a natural reading. But as clang told you that's not what it does, and if you look at the class semantics it doesn't make sense either.

**Pull request merged**